### ISSUE TITLE
Can't get good performance.

### ISSUE THREAD

**Issue by EurFelux** [uark-cviu/Micron-BERT#6]
Is it possible to provide a classification model for the recognition task? I tested the results on CASME2, using features extracted from a pre-trained model for global pooling (also tried to extract CLS tokens), hooked to an MLP classifier, but both performed poorly. Accuracy is lower than 60%.

**Comment by fw-codes-files** [comment]
> Is it possible to provide a classification model for the recognition task? I tested the results on CASME2, using features extracted from a pre-trained model for global pooling (also tried to extract CLS tokens), hooked to an MLP classifier, but both performed poorly. Accuracy is lower than 60%.

同学你好，请问一下怎么得到casme2数据集？

**Comment by wangz315** [comment]
> 

Did you use forward_encoder function to get the features?
what I notice is that diag_att is set to False in pretrained weight. I'm not sure is the released weights trained with DMA.

And, I'm also not sure x and x_ref, which one should be onset frame. My guess is x should be apex frame, since both notated as I/t+d. Or maybe it really does not matter...

**Comment by ilikecodeing** [comment]
> 
Is it possible to provide a code about features extracted  ? because I am very confused about it

**Comment by EurFelux** [comment]
> > Is it possible to provide a classification model for the recognition task? I tested the results on CASME2, using features extracted from a pre-trained model for global pooling (also tried to extract CLS tokens), hooked to an MLP classifier, but both performed poorly. Accuracy is lower than 60%.是否可以为识别任务提供分类模型？我在 CASME2 上测试了结果，使用从预训练的全局池化模型中提取的特征（也尝试提取 CLS 令牌），挂在 MLP 分类器上，但两者都表现不佳。准确度低于60%。
> 
> 同学你好，请问一下怎么得到casme2数据集？

@fw-codes-files  如果你还没有解决，请访问 http://casme.psych.ac.cn/casme/c2 按照指引申请数据集。

**Comment by EurFelux** [comment]
> > 
> 
> Did you use forward_encoder function to get the features? what I notice is that diag_att is set to False in pretrained weight. I'm not sure is the released weights trained with DMA.
> 
> And, I'm also not sure x and x_ref, which one should be onset frame. My guess is x should be apex frame, since both notated as I/t+d. Or maybe it really does not matter...

@wangz315 To apply `att`, use `forward_encoder` directly. The `att` is trained, and you can see this if you plot a heat map.

I input onset frame and apex frame as `x` and `x_ref`, respectively.

**Comment by EurFelux** [comment]
> > 
> 
> Is it possible to provide a code about features extracted ? because I am very confused about it

@ilikecodeing  I didn't end up using Micron-BERT in the code of my project, so my code about that part is gone. You could refer [micron_bert.py](https://github.com/uark-cviu/Micron-BERT/blob/main/micron_bert.py).
```python
# To extract features
image = load_image(your_image_path)
image_tensor = image_to_tensor(image)
with torch.no_grad():
    features = model.extract_features(image)
    # Use this features for finetunning.
```

**Comment by bugger69** [comment]
> Is it possible to provide a classification model for the recognition task? I tested the results on CASME2, using features extracted from a pre-trained model for global pooling (also tried to extract CLS tokens), hooked to an MLP classifier, but both performed poorly. Accuracy is lower than 60%.

Uhh could you provide me the pre trained model along with the code for testing? It would be a great help if you could since I need it for a project asap...

**Comment by tony19980810** [comment]
> Is it possible to provide a classification model for the recognition task? I tested the results on CASME2, using features extracted from a pre-trained model for global pooling (also tried to extract CLS tokens), hooked to an MLP classifier, but both performed poorly. Accuracy is lower than 60%.
同学你好，请问问题解决了吗

**Comment by ShannonZ** [comment]
I cannot get the reported performance using the pre-trained model ('CASME2-is224-p8-b16-ep200.pth').  Accuracy is lower than 60% on CASME II.
